src/pytorch_lightning/_graveyard/cloud_io.py

```diff
@@ -0,0 +1,23 @@
+import pickle
+import warnings
+from copy import deepcopy
+from typing import Any
+
+
+class RedirectingUnpickler(pickle.Unpickler):
+    def find_class(self, module: str, name: str) -> Any:
+        if module.startswith("pytorch_" + "lightning"):
+            # for the standalone package this won't do anything,
+            # for the unified mirror package it will redirect the imports
+
+            old_module = deepcopy(module)
+            module = "pytorch_lightning" + module[len("pytorch_" + "lightning") :]
+
+            # this warning won't trigger for standalone as these imports are identical
+            if module != old_module:
+                warnings.warn(f"Redirecting import of {module}.{name} to pytorch_lightning.{name}")
```

Comment:
What is the motivation for showing this to the user? Maybe we could say what they should do to avoid the warning?

Reply:
note that in a mono-package it will be replaced to be
```
warnings.warn(f"Redirecting import of {module}.{name} to lightning.pytorch.{name}")
```

Reply:
They cannot do much to prevent this. The issue is that we did pickle entire classes in old checkpoints which results in pytorch_lightning being imported (or failure to load the checkpoint at all if PL is not present standalone). This warning is just to be on the safe side, that users will be informed what we are doing. They also should only experience this while loading legacy checkpoints